### PR TITLE
feat: Add summary property to calcite-flow-item

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #450

## Summary

Add summary property to calcite-flow-item

### CODE REVIEW

**Comment by AdelheidF** [comment]
Issue: https://github.com/Esri/calcite-app-components/issues/450

**Comment by driskull** [comment]
Thanks @AdelheidF!

@asangma this just needs CSS love.

**Comment by asangma** [comment]
@driskull Do you think we should do this on Panel instead?

File: src/components/calcite-flow-item/calcite-flow-item.tsx [hunk 1]
```diff
@@ -52,6 +53,11 @@ export class CalciteFlowItem {
    */
   @Prop() showBackButton = false;
 
+  /**
+   * Summary text.
```
Comment (pr3tori4n) [inline src/components/calcite-flow-item/calcite-flow-item.tsx]: pebble: Needs better description. Curious what the subtle difference between heading and summary are, as a summary can act as a heading like it does in our PRs sometimes.

File: src/components/calcite-flow-item/calcite-flow-item.e2e.ts [hunk 2]
```diff
@@ -67,6 +67,16 @@ describe("calcite-flow-item", () => {
...
   });
 
+  it("should have default summary", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<calcite-flow-item summary="test"></calcite-flow-item>');
+
+    const element = await page.find(`calcite-flow-item >>> .${CSS.summary}`);
+
+    expect(element).toEqualText("test");
```
Comment (pr3tori4n) [inline src/components/calcite-flow-item/calcite-flow-item.e2e.ts]: Suggestion to move the string to a TEXT content in resources, so updates to components keep the tests in sync.

**Comment by driskull** [comment]
> Do you think we should do this on Panel instead?

@asangma we could, if we want to restrict what users put into a panel as having a heading/summary in the middle and nothing else. Do you want to go that route?

File: src/components/calcite-flow-item/calcite-flow-item.e2e.ts (hunk 2 above)
Comment (driskull) [reply src/components/calcite-flow-item/calcite-flow-item.e2e.ts]: Its local to this file so it wouldn't get out of sync.

**Comment by asangma** [comment]
@driskull Based on our conversation and the fact that Adelheid doesn't need this anytime soon, do we want to move it to the next milestone?

**Comment by driskull** [comment]
Yup.  moved

**Comment by driskull** [comment]
Alan this just needs styling I believe.

**Comment by driskull** [comment]
@asangma merged in latest master, resolved conflicts. Closing this PR. Can you just keep yours open and target master?

**Pull request closed without merging**